scripts/remove_work/run.py

```diff
@@ -0,0 +1,300 @@
+        assert len(app_containers) == 1
+
+        app_env_vars = {e["name"]: e["value"] for e in app_containers[0]["environment"]}
+
+        app_secrets = {s["name"]: s["valueFrom"] for s in app_containers[0]["secrets"]}
+
+        for name, value_from in app_secrets.items():
+            resp = ssm_client.get_parameter(Name=value_from, WithDecryption=True)
+            app_secrets[name] = resp["Parameter"]["Value"]
+
+        # 3. Once we have the config and password, we can remove the work from
+        # the Elasticsearch index (if present).
+        es_host = "%s://%s:%s/" % (
+            app_secrets["es_protocol"],
+            app_secrets["es_host"],
+            app_secrets["es_port"],
+        )
+
+        for index_name in (app_env_vars["es_index_v1"], app_env_vars["es_index_v2"]):
+            print("··· Looking up %s in index %s" % (catalogue_id, index_name))
+            resp = requests.get(
+                f"{es_host}{index_name}/_doc/{catalogue_id}",
+                auth=es_auth,
+            )
+
+            if resp.status_code == 404:
+                print("··· Work is not in this index, skipping")
+                continue
+
+            assert resp.status_code == 200, resp.json()
+
+            existing_work = resp.json()["_source"]
+
+            if existing_work["type"] == "IdentifiedInvisibleWork":
+                print(
+                    "··· Work is already suppressed as an IdentifiedInvisibleWork, skipping"
+                )
+                continue
+
+            # While we're looking at API responses, try to get the Miro ID.
+            identifiers = [existing_work["sourceIdentifier"]] + existing_work[
+                "otherIdentifiers"
+            ]
+            miro_identifiers = [
+                idf
+                for idf in identifiers
+                if idf["identifierType"]["id"] == "miro-image-number"
+            ]
+            assert len(miro_identifiers) == 1
+            miro_identifier = miro_identifiers[0]["value"]
+            if miro_id is None:
+                miro_id = miro_identifier
+            else:
+                assert miro_id == miro_identifier, "Multiple Miro IDs? %s, %s" % (
+                    miro_id,
+                    miro_identifier,
+                )
+
+            assert existing_work["type"] == "IdentifiedWork"
+
+            new_work = {
+                "canonicalId": existing_work["canonicalId"],
+                "sourceIdentifier": existing_work["sourceIdentifier"],
+                "type": "IdentifiedInvisibleWork",
+                # We bump the version so any in-flight works won't overwrite
+                # this one.
+                "version": existing_work["version"] + 1,
+            }
+            print("··· Replacing work with an IdentifiedInvisibleWork")
+            resp = requests.put(
+                "%s%s/%s/%s" % (es_host, index_name, index_name, catalogue_id),
```

Comment:
Not sure what version of python people are using (I've on 3.7), but using the [new string interpolation](https://www.python.org/dev/peps/pep-0498/) introduced in 3.6 makes these sorts of expressions much more readable IMO:

```python
resp = requests.put(
    f"{es_host}{index_name}/{index_name}/{catalogue_id}"
    ...
)
```